be/src/exec/parquet_builder.cpp

```diff
@@ -0,0 +1,351 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "parquet_builder.h"
+
+#include <arrow/buffer.h>
+#include <arrow/io/file.h>
+#include <arrow/io/interfaces.h>
+#include <parquet/arrow/writer.h>
+#include <parquet/exception.h>
+
+#include "column/chunk.h"
+#include "column/column_helper.h"
+#include "common/logging.h"
+#include "exprs/column_ref.h"
+#include "exprs/expr.h"
+#include "runtime/exec_env.h"
+#include "util/priority_thread_pool.hpp"
+
+namespace starrocks {
+
+ParquetOutputStream::ParquetOutputStream(std::unique_ptr<WritableFile> writable_file)
+        : _writable_file(std::move(writable_file)) {
+    set_mode(arrow::io::FileMode::WRITE);
+}
+
+ParquetOutputStream::~ParquetOutputStream() {
+    arrow::Status st = ParquetOutputStream::Close();
+    if (!st.ok()) {
+        LOG(WARNING) << "close parquet output stream failed, err msg: " << st.ToString();
+    }
+}
+
+arrow::Status ParquetOutputStream::Write(const std::shared_ptr<arrow::Buffer>& data) {
+    Write(data->data(), data->size());
```

Comment:
Should check status of the return result.

Reply:
done